### DISCUSSION TITLE
Is there any way to make the web3modal open with only the qrcode?

### DISCUSSION THREAD

**Discussion by dohankk** [WalletConnect/walletconnect-monorepo#3776] [Developer Support]
```ts
const wagmiConfig = createConfig({
			autoConnect: false,
			connectors: [
				new WalletConnectConnector({ chains: this.chains, options: { projectId: this.#projectId, showQrModal: false, metadata: this.metadata, showQrModal: true } }),
				ic,
				new CoinbaseWalletConnector({ chains: this.chains, options: { appName: this.metadata.name } })
			]
		});

		const modal = createWeb3Modal({ wagmiConfig, projectId: this.#projectId, chains: this.chains, themeMode: 'light' });

...
modal.open();
```

In this code, it looks like the picture below.
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/106723173/edf4275d-afa3-48a0-9d11-7880dd687a6d)

Is there any way to make the web3modal open with only the qrcode?

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/106723173/014839dd-62ab-42fa-8e03-92ac7b5bef8a)

**Comment by dohankk** [comment]
I found `EthereumProvider`.

```ts
const provider = await EthereumProvider.init({
			projectId: this.#projectId,
			showQrModal: true,
			chains: this.chains,
			metadata: this.metadata
		});
		
		...
		await provider.connect();
```

The qrcode modal is not closing after taking a qrcode with a metamask and approving it.

`provider.on("connect", () => {
 console.log("connected");
});`

Even if I run the above code, the console is not closed.

**Reply by glitch-txs** [reply]
Hi you can follow this example for ethereum-provider
https://github.com/WalletConnect/web-examples/tree/main/basic/dapps/ethereum-provider

**Comment by fukemy** [comment]
hi, any solution?

**Comment by Nipunc55** [comment]
You have create own connect wallet button, and show only wallet connect connecter.
````
export function WalletOptions() {
  const { connectors, connect } = useConnect()

  return connectors.map((connector) => (
    <WalletOption
      key={connector.uid}
      connector={connector}
      onClick={() => connect({ connector })}
    />
  ))
}
````

I you your config  pass only walletConnect option

````
export const config = createConfig({
  chains: [mainnet, base],
  connectors: [
   
    walletConnect({ projectId }), //pass only walletConnect option

  ],
  transports: {
    [mainnet.id]: http(),
    [base.id]: http(),
  },
})
````   

[refer this for more details](https://wagmi.sh/react/guides/connect-wallet)
